aten/src/ATen/native/transformers/attention.cpp

```diff
@@ -819,6 +838,8 @@ std::tuple<Tensor, Tensor> _scaled_dot_product_attention_math(
           new_attn_mask.masked_fill_(attn_mask->logical_not(), -std::numeric_limits<double>::infinity());
           attn_mask = new_attn_mask;
         }
+        TORCH_CHECK(attn_mask->dtype() == query.dtype(),
```

Comment:
I'd move this into `validate_sdpa_input` function. This seems fragile. There's an implicit assumption the mask will be converted to match the dtype of the query if it's bool.